eureka2-write-server/src/main/java/com/netflix/eureka2/server/channel/SenderReplicationChannel.java

```diff
@@ -43,11 +43,8 @@ public SenderReplicationChannel(TransportClient client, ReplicationChannelMetric
 
     @Override
     public Observable<ReplicationHelloReply> hello(final ReplicationHello hello) {
-        if (state.get() == STATE.Closed) {
-            return Observable.error(CHANNEL_CLOSED_EXCEPTION);
-        }
-        if (!state.compareAndSet(STATE.Idle, STATE.Handshake)) {
-            return Observable.error(HANDSHAKE_FINISHED_EXCEPTION);
+        if (!moveToState(STATE.Idle, STATE.Handshake)) {
```

Comment:
The state transition is driven by the client. This guard enforces single handshake that happens in the very beginning. We could add explicit channel close in case of failure, but we do not do this on the client side, and let the client handle the errors. We drift towards more "standardized" state machine in the channel. Lets revisit how we should handle it, including lifecycle/shutdown handling.